synapse_rock/rockcraft.yaml

```diff
@@ -10,176 +10,219 @@ build-base: ubuntu@24.04
+      /rust/toolchains/$RUST_VERSION-x86_64-unknown-linux-gnu/bin/rustc -V > $CRAFT_PART_INSTALL/rust-version
+      pip3 install -U pip setuptools
+      pip3 install --root-user-action=ignore "poetry==$POETRY_VERSION"
+      cp pyproject.toml poetry.lock /synapse/
+      /usr/local/bin/poetry export --extras all -o /synapse/requirements.txt
+      pip3 install --prefix="/install" --no-deps --no-warn-script-location -r /synapse/requirements.txt
+      git clone https://github.com/canonical/synapse_stats_exporter.git
+      cd synapse_stats_exporter
+      pip3 install --prefix="/install" --no-warn-script-location -U .
+      cd ..
+      git clone --branch v1.4.0 --depth 1 https://github.com/matrix-org/synapse-s3-storage-provider
+      cd synapse-s3-storage-provider
+      pip3 install --prefix="/install" --no-warn-script-location -U .
+      cd ..
+      cp -r synapse /synapse/
+      cp -r rust /synapse/
+      cp pyproject.toml README.rst build_rust.py Cargo.toml Cargo.lock /synapse/
+      pip3 install --prefix="/install" --no-deps --no-warn-script-location /synapse[all];
+      git clone https://git.buechner.me/nbuechner/synapse-invite-checker
+      cd synapse-invite-checker
+      git checkout ab6c8b78d78c4cbf31e1a30981ae45c09285b34a
+      pip3 install --prefix="/install" --no-warn-script-location -U .
+      cd ..
+      # fix issue while creating file
+      # https://github.com/element-hq/synapse/issues/17882
+      pip3 install --prefix="/install" --force-reinstall -v "Twisted==24.7.0"
+      cp docker/start.py $CRAFT_PART_INSTALL/
+      chmod 755 $CRAFT_PART_INSTALL/start.py
+      cp -r docker/conf $CRAFT_PART_INSTALL/
+      cp -r /usr/local $CRAFT_PART_INSTALL/usr/
+      cp -r /install/local/* $CRAFT_PART_INSTALL/usr/local/
+      mkdir -p $CRAFT_PART_INSTALL/usr/local/attributemaps
+      chmod 755 $CRAFT_PART_INSTALL/usr/local/attributemaps
+      rm -rf $CRAFT_PART_INSTALL/aws/dist/awscli/examples/
+    overlay-packages:
+      - ca-certificates
+      - libjemalloc2
+  patches:
+    plugin: dump
+    after: [synapse]
+    source: patches
+    organize:
+      "*": srv/patches/
+  patch-start:
+    after:
+      - patches
+    plugin: nil
+    build-packages:
+      - git
+    override-stage: |
+      patch -f -u ./start.py -i srv/patches/start.patch
+  synapse-conf:
+    plugin: dump
+    source: attributemaps
+    organize:
+      login_ubuntu.py: /usr/local/attributemaps/login_ubuntu.py
+  mas-templates:
+    plugin: dump
+    <<: *mas-source
+    source-subdir: templates
+    organize:
+      "*": mas/share/templates/
+  mas-translations:
+    plugin: dump
+    <<: *mas-source
+    source-subdir: translations
+    organize:
+      "*": mas/share/translations/
+  mas-assets:
+    plugin: nil
+    <<: *mas-source
+    source-subdir: frontend
+    build-environment:
+      - NODE_URI: "https://nodejs.org/dist/v22.11.0/node-v22.11.0-linux-x64.tar.gz"
```

Comment:
Same question here about a different version
https://github.com/element-hq/matrix-authentication-service/blob/24be677a0669c212bc5dec008968e5af79165248/Dockerfile#L12

Reply:
Thanks, I replied to that comment